### PR TITLE
Update Native Auth password reset API response

### PR DESCRIPTION
**Pull request by rmccahill**
This PR updates the relevant response classes to include the continuation token returned from the updated poll completion response, and updates constructors in test classes to reflect the change. 

Note: I've also updated the test runner to Robolectric from PowerMock for the reset password test class to be in line with other test cases, as it was causing test failures locally.

### CODE REVIEW

File: common/src/test/java/com/microsoft/identity/common/internal/providers/microsoft/nativeauth/integration/ResetPasswordOAuth2StrategyTest.kt [hunk 1]
```diff
@@ -81,7 +81,7 @@ import java.util.UUID
 
 
 @RunWith(
-    PowerMockRunner::class
+    RobolectricTestRunner::class
```
Comment (SammyO) [inline common/src/test/java/com/microsoft/identity/common/internal/providers/microsoft/nativeauth/integration/ResetPasswordOAuth2StrategyTest.kt:84]: Why this change?

Comment (rmccahill) [reply common/src/test/java/com/microsoft/identity/common/internal/providers/microsoft/nativeauth/integration/ResetPasswordOAuth2StrategyTest.kt:84]: From the description above: 

> I've also updated the test runner to Robolectric from PowerMock for the reset password test class to be in line with other test cases, as it was causing test failures locally.

File: changelog.txt [hunk 2]
```diff
@@ -1,6 +1,7 @@
 V.Next
 ---------
 - [MINOR] Implement updates of the native auth web API (#2261)
+- [MINOR] Implement update of the native auth web password reset API response (#2281)
```
Comment (Yuki-YuXin) [inline changelog.txt]: I personally feel that this description is part of the previous one. Could we use more specific words like
"Update ResetPasswordPollCompletion related command, results, response to supports automatic login after password reset."

Comment (rmccahill) [reply changelog.txt]: Good point - I've updated the description to better reflect the changes

**Review by Yuki-YuXin — APPROVED** [review]

**Review by SammyO — APPROVED** [review]
